instrumentation/kamon-elasticsearch/src/main/scala/kamon/instrumentation/elasticsearch/ESInstrumentation.scala

```diff
@@ -0,0 +1,41 @@
+package kamon.instrumentation.elasticsearch
+
+import kamon.Kamon
+import kamon.instrumentation.elasticsearch.highlevel.{AsyncClientInstrumentation, SyncClientInstrumentation}
+import kamon.trace.Span
+import kanela.agent.api.instrumentation.InstrumentationBuilder
+import org.apache.http.HttpEntity
+import org.elasticsearch.client.{Response, ResponseListener}
+
+class ESInstrumentation extends InstrumentationBuilder {
+  onType("org.elasticsearch.client.RestClient")
+    .advise(method("performRequestAsync").and(takesArguments(2)), classOf[AsyncRestClientInstrumentation])
+    .advise(method("performRequest").and(takesArguments(1)), classOf[SyncRestClientInstrumentation])
+
+  onType("org.elasticsearch.client.RestHighLevelClient")
+    .advise(method("internalPerformRequest").and(takesArguments(5)), classOf[SyncClientInstrumentation])
+    .advise(method("internalPerformRequestAsync").and(takesArguments(6)), classOf[AsyncClientInstrumentation])
+}
+
+class InstrumentedListener(inner: ResponseListener, span: Span) extends ResponseListener {
+  override def onSuccess(response: Response): Unit = {
+    span.finish()
+    inner.onSuccess(response)
+  }
+
+  override def onFailure(exception: Exception): Unit = {
+    span.fail(exception)
+    inner.onFailure(exception)
+  }
+}
+
+object RequestSizeHistogram {
+  private val histogram = Kamon.histogram("elastic.request.size").withoutTags()
+
+  def record(entity: HttpEntity): Unit = {
+    Option(entity)
+      .map(_.getContentLength)
+      .filter(_ >= 0)
```

Comment:
In what cases could the content be negative or zero?

Reply:
Just being cautious. From the docs:

```
     * @return  the number of bytes of the content, or
     *          a negative number if unknown. If the content length is known
     *          but exceeds {@link java.lang.Long#MAX_VALUE Long.MAX_VALUE},
     *          a negative number is returned.
```